services/cd-service/pkg/repository/repository.go

```diff
@@ -2188,7 +2187,23 @@ func (s *State) GetCurrentApplicationLocks(ctx context.Context) (db.AllAppLocks,
 	return result, nil
 }
 
-func (s *State) GetApplicationReleases(application string) ([]uint64, error) {
+func (s *State) GetAllApplicationReleases(ctx context.Context, transaction *sql.Tx, application string) ([]uint64, error) {
+	if s.DBHandler.ShouldUseOtherTables() {
+		app, err := s.DBHandler.DBSelectAllReleasesOfApp(ctx, transaction, application)
+		if err != nil {
+			return nil, fmt.Errorf("could not get all releases of app %s: %v", application, err)
+		}
+		if app == nil {
+			return nil, fmt.Errorf("could not get all releases of app %s (nil)", application)
+		}
+		res := ptr.ToUint64Slice(app.Metadata.Releases)
+		return res, nil
+	} else {
+		return s.GetAllApplicationReleasesFromFile(application)
+	}
+}
+
+func (s *State) GetAllApplicationReleasesFromFile(application string) ([]uint64, error) {
```

Comment:
We call such function `getSomethingFromManifest` elsewhere. Is there any difference between `FromManifest` and `FromFile`? If not, let's rename this.

Reply:
done